### PR TITLE
Improved parsing

### PR DESCRIPTION
**Pull request by Archer4499**
It should parse all the known cases properly now, and retry up to 3 times on a timeout or invalid data.

I haven't tested the retry code properly yet because it's night here. And I'm not sure if it's the best way of doing it, I'm happy for any input on this.

There's some commented out code as a base for getting multiple inverters and meter data if wanted in the future.

I added a few more test cases, but I wasn't sure how to test exceptions so I didn't add any of testing that.

I added the data and example tables to the readme as well.

You could maybe add some of the extra info as attributes to the entities in home assistant. I think the connection status could be useful at least.

### CODE REVIEW

**Review by kvanzuijlen — COMMENTED** [review]
I'd have to look at this closer at a later moment but thank you for another contribution!

First thoughts;
It might be more readable to use [opnieuw](https://pypi.org/project/opnieuw/) or [retry2](https://pypi.org/project/retry2/) for the retry functionality.

File: tests/integration_test.py [hunk 1]
```diff
@@ -17,6 +17,24 @@
         3187,
```
[the remark this answers is not in the document]
Comment (Archer4499) [reply tests/integration_test.py:17]: Done.
I'm not really sure why I didn't before.

**Review by kvanzuijlen — APPROVED** [review]

**Comment by kvanzuijlen** [comment]
LGTM, I'm going to further improve it with more tests etc in a separate PR

**Comment by Archer4499** [comment]
I've realised that it doesn't actually wait when retrying because of a data error, just timeouts because of the timeout time.
I think making it all async and using asyncio.sleep() would be best and then use aiohttp instead of requests, which shouldn't be difficult for this simple use. HA requires async and aiohttp for their platinum integration rating, so it seems to be the preferred one to use.
I think it would just require changing the calls to this in the integration to call directly instead of with hass.async_add_executor_job.

Also, I think the ZeverSolarInvalidData exception needs adding to the handling in config_flow.py, and maybe elsewhere?

**Pull request merged**